site/src/pages/CreateUserPage/CreateUserForm.tsx

```diff
@@ -104,6 +114,10 @@ export const CreateUserForm: FC<
 		error,
 	);
 
+	useEffect(() => {
+		onPasswordChange?.(form.values.password);
+	}, [form.values.password, onPasswordChange]); // Run effect when password changes
```

Comment:
I think we want to avoid `useEffect` as much as possible - it causes unexpected rerenders, which, decoupled with your debouncing function, are going to be difficult to troubleshoot. Instead, I suggest you add an `onChange` attribute to the `TextField` component used for the password input. Something like:
```
onChange={(e) => {
    form.handleChange(e);
    handlePasswordChange(e.target.value);
}}
```

If you search in the file you can see us using this pattern for other inputs.